### PR TITLE
Token expiry updates

### PR DESCRIPTION
**Pull request by monstermunchkin**
Related to https://github.com/lxc/lxd/issues/10505

### CODE REVIEW

**Review by tomponline — CHANGES_REQUESTED** [review]
Please can you include the missing API doc refresh commit in this too, thanks.

See https://github.com/lxc/lxd/pull/11037/files#diff-216fa85d6fc7c80b69a6c22e22f0bd383806df995d6801b67ba4990eacff9e55R381

**Review by tomponline — APPROVED** [review]

**Comment by tomponline** [comment]
@monstermunchkin I just thought, in https://github.com/lxc/lxd/pull/11038, do we need something to prune tokens that have expired?

**Comment by monstermunchkin** [comment]
> @monstermunchkin I just thought, in #11038, do we need something to prune tokens that have expired?

Oh damn, yeah, I totally forgot about that. I'll add it to this PR.

**Comment by tomponline** [comment]
> > @monstermunchkin I just thought, in #11038, do we need something to prune tokens that have expired?
> 
> Oh damn, yeah, I totally forgot about that. I'll add it to this PR.

Is there some way we can combine it with the cluster token removal so we dont have 2 tasks?

**Comment by monstermunchkin** [comment]
> Is there some way we can combine it with the cluster token removal so we dont have 2 tasks?

That shouldn't be a problem. We can rename it to `autoRemoveExpiredTokens` which then handles remote tokens, and cluster tokens if clustered.

File: lxd/tokens.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		err = op.Start()
+		if err != nil {
+			logger.Error("Failed to remove expired tokens", logger.Ctx{"err": err})
+			return
+		}
+
+		_, _ = op.Wait(ctx)
+	}
+
+	return f, task.Hourly()
```
Comment (tomponline) [inline lxd/tokens.go:98]: This is hourly, but comment above says daily.

File: lxd/daemon.go [hunk 2]
```diff
@@ -1591,6 +1591,9 @@ func (d *Daemon) init() error {
 
 		// Auto-renew server certificate (daily)
 		d.tasks.Add(autoRenewCertificateTask(d))
+
+		// Remove expired tokens (daily)
+		d.clusterTasks.Add(autoRemoveExpiredTokensTask(d))
```
Comment (tomponline) [inline lxd/daemon.go:1596]: daily or hourly?

File: lxd/tokens.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+func autoRemoveExpiredTokens(ctx context.Context, d *Daemon) error {
+	removeClusterJoinTokens := false
+
+	clustered, err := cluster.Enabled(d.db.Node)
+	if err != nil {
+		return err
+	}
+
+	// If clustered, remove the cluster join tokens as well.
```
Comment (tomponline) [inline lxd/tokens.go:22]: > // If cluster leader,...

**Review by tomponline — APPROVED** [review]

**Comment by monstermunchkin** [comment]
@tomponline please don't merge yet. I'm still running tests.

**Comment by monstermunchkin** [comment]
@tomponline good to go.

**Pull request merged**